### ISSUE TITLE
MalformedInputException

### ISSUE THREAD

**Issue by fommil** [sbt/sbt-header#56]
when I upgraded sbt-header to the latest in ensime-server I got this exception.

```
[error] (monkeys/compile:createHeaders) java.nio.charset.MalformedInputException: Input length = 1
```

should be easy to reproduce since everything is available on github.

Downgrading as workaround...

**Comment by andyscott** [comment]
I am running into the same issue when our builds run on CI. Downgrading worked as a workaround.

**Comment by imcharsi** [comment]
Hi.

At first, My English is not good. I'm sorry about that.

In my case, I see this error in Windows 10. But In Linux, It's fine without any modification.

I goggled about this and I think I found a solution(Thanks so much who wrote this).
Please refer to [here](http://stackoverflow.com/questions/13625024/how-to-read-a-text-file-with-mixed-encodings-in-scala-or-java).
I inserted below code into [HeaderCreator.scala:29](https://github.com/sbt/sbt-header/blob/master/src/main/scala/de/heikoseeberger/sbtheader/HeaderCreator.scala#L29).

```
  implicit val codec = Codec("UTF-8")
  codec.onMalformedInput(CodingErrorAction.REPLACE)
  codec.onUnmappableCharacter(CodingErrorAction.REPLACE)
  val (firstLine, text) = scala.io.Source.fromInputStream(input)(codec).mkString match {
...
```

I tested this and checked that this works.
I hope this can help you

Thanks for read.

**Comment by britter** [comment]
@fommil I wonder whether this is related to #58 

@imcharsi thank you for your analysis. I'll have a look and see what we can do.

**Comment by fommil** [comment]
still getting this with sbt-header 1.8.0 on our CI, but not locally. very weird.

**Comment by britter** [comment]
@fommil I suspect your CI server is running a different OS? Which one?

**Comment by fommil** [comment]
@britter docker running Debian Jessie

**Comment by britter** [comment]
@fommil should be easy to reproduce if you're running docker, but I don't have the cycles at the moment to work on this. I'll try to have a look when I have some more time for sat-header again.

**Comment by julianheres** [comment]
Ran into this on Ubuntu Server 17.04. Sorted it out by re-configuring the locales: 'dpkg-reconfigure locales' and setting es_CO UTF-8 (my locale) in addition to en_US UTF-8 (US Locale). Hope it helps!

Cheers.

**Comment by ignasi35** [comment]
**TL;DR** the default encoding picked by the JVM executing `sbt` is not your file encoding. Test using `System.getProperty("file.encoding")`.


We use `1.8.0` in Lagom and everything worked like a charm. Today I upgraded IntelliJ from 2017.1.x to 2017.2.5 (on an MBP) and then upgraded the scala plugin in IntelliJ to 2017.2.11 which includes support for sbt 1.0.

After opening the Lagom project in the upgraded IntelliJ, any time  I run `sbt createHeaders` from the embedded terminal in IntelliJ-terminal plugin I get `[error] (testkit-javadsl/compile:createHeaders) java.nio.charset.MalformedInputException: Input length = 1` errors for several in several projects.

If I run `sbt createHeaders` from iTerm2 everything works fine.

Turns out the updated plugin in IntelliJ was resolving `LC_CTYPE="C"` causing the JVM to pick the `file.encoding` == `US-ASCII`. Setting `LC_CTYPE="UTF-8"` on my `.bash_profile` solved the issue in all my run environments.

PS: The issue had nothing to do with the scala plugin upgrade or sbt 1.0.

**Comment by britter** [comment]
@hseeberger maybe we should put a section about this problem into README.md.

**Comment by hseeberger** [comment]
Happy to accept a PR.

**Comment by nishanpatel** [comment]
I don't know whether this is plugin issue or sbt issue. I tried all possible solutions found in google but I am still facing the same issue. 

java.nio.charset.UnmappableCharacterException: Input length = 1
        at java.nio.charset.CoderResult.throwException(CoderResult.java:282)
        at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
        at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
        at java.io.InputStreamReader.read(InputStreamReader.java:184)
        at java.io.BufferedReader.fill(BufferedReader.java:161)
        at java.io.BufferedReader.read(BufferedReader.java:182)
        at scala.io.BufferedSource$$anonfun$iter$1$$anonfun$apply$mcI$sp$1.apply$mcI$sp(BufferedSource.scala:38)
        at scala.io.Codec.wrap(Codec.scala:68)
        at scala.io.BufferedSource$$anonfun$iter$1.apply(BufferedSource.scala:38)
        at scala.io.BufferedSource$$anonfun$iter$1.apply(BufferedSource.scala:38)
        at scala.collection.Iterator$$anon$9.next(Iterator.scala:162)
        at scala.collection.Iterator$$anon$17.hasNext(Iterator.scala:511)
        at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
        at scala.io.Source.hasNext(Source.scala:226)
        at scala.collection.Iterator$class.foreach(Iterator.scala:727)
        at scala.io.Source.foreach(Source.scala:178)
        at scala.collection.TraversableOnce$class.addString(TraversableOnce.scala:320)
        at scala.io.Source.addString(Source.scala:178)
        at scala.collection.TraversableOnce$class.mkString(TraversableOnce.scala:286)
        at scala.io.Source.mkString(Source.scala:178)
        at scala.collection.TraversableOnce$class.mkString(TraversableOnce.scala:288)
        at scala.io.Source.mkString(Source.scala:178)
        at scala.collection.TraversableOnce$class.mkString(TraversableOnce.scala:290)
        at scala.io.Source.mkString(Source.scala:178)
        at de.heikoseeberger.sbtheader.HeaderCreator.<init>(HeaderCreator.scala:30)
        at de.heikoseeberger.sbtheader.HeaderCreator$.apply(HeaderCreator.scala:65)
        at de.heikoseeberger.sbtheader.HeaderPlugin$.de$heikoseeberger$sbtheader$HeaderPlugin$$createHeader(HeaderPlugin.scala:136)
        at de.heikoseeberger.sbtheader.HeaderPlugin$$anonfun$1$$anonfun$apply$2.apply(HeaderPlugin.scala:127)
        at de.heikoseeberger.sbtheader.HeaderPlugin$$anonfun$1$$anonfun$apply$2.apply(HeaderPlugin.scala:127)
        at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:251)
        at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:251)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:251)
        at scala.collection.AbstractTraversable.flatMap(Traversable.scala:105)
        at de.heikoseeberger.sbtheader.HeaderPlugin$$anonfun$1.apply(HeaderPlugin.scala:127)
        at de.heikoseeberger.sbtheader.HeaderPlugin$$anonfun$1.apply(HeaderPlugin.scala:127)
        at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:251)
        at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:251)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:251)
        at scala.collection.AbstractTraversable.flatMap(Traversable.scala:105)
        at de.heikoseeberger.sbtheader.HeaderPlugin$.de$heikoseeberger$sbtheader$HeaderPlugin$$createHeadersTask(HeaderPlugin.scala:127)
        at de.heikoseeberger.sbtheader.HeaderPlugin$$anonfun$toBeScopedSettings$3.apply(HeaderPlugin.scala:103)
        at de.heikoseeberger.sbtheader.HeaderPlugin$$anonfun$toBeScopedSettings$3.apply(HeaderPlugin.scala:103)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[error] (coreJVM/compile:createHeaders) java.nio.charset.UnmappableCharacterException: Input length = 1
[error] Total time: 1 s, completed 19-Sep-2018 17:04:56

**Comment by britter** [comment]
@nishanpatel can you please provide a minimal sample project for reproducing the error?

**Comment by fommil** [comment]
better yet, please create a fresh ticket.

**Comment by nishanpatel** [comment]
> @nishanpatel can you please provide a minimal sample project for reproducing the error?

I am using below project to run in my local machine with following configurations, Windows 10, Scala version 2.12.6, java version 1.8.0_161 and SBT version 1.2.1.

[Project](https://github.com/scala-exercises/scala-exercises)

**Comment by nishanpatel** [comment]
Hi @britter @fommil thanks for your time. I have created new issue with all the details.

[UnmappableCharacterException Exception](https://github.com/sbt/sbt-header/issues/160)
